src/utils/update/managing.officer.mapper.ts

```diff
@@ -88,3 +88,11 @@ export const mapMoPrivateAddress = (moPrivateData: ManagingOfficerPrivateData[],
     }
   }
 };
+
+export const mapMoPrivateDOB = (moPrivateData: ManagingOfficerPrivateData[], ch_reference: string) => {
+  for (const managingOfficerData of moPrivateData) {
```

Comment:
We could combine both the `mapMoPrivateAddress` and `mapMoPrivateDOB` functions into one to avoid duplication, something like:

```
const mapManagingOfficerIndividualData = (managingOfficer: ManagingOfficerIndividual, moPrivateData) => {
  for (const managingOfficerData of moPrivateData) {
    if (managingOfficerData.hashedId === managingOfficer.ch_reference) {
      managingOfficer.usual_residential_address = mapBOMOAddress(managingOfficerData.residentialAddress);
      managingOfficer.date_of_birth = mapInputDate(managingOfficerData.dateOfBirth);
    }
  }
};
```

Reply:
Just one observation which may or may not help. We need to rely on having a day of birth so that the backend displays data correctly (2 open bugs). So may end up failing if no day found - which would make the day kind of special.

Reply:
> We could combine both the `mapMoPrivateAddress` and `mapMoPrivateDOB` functions into one to avoid duplication, something like:
> 
> ```
> const mapManagingOfficerIndividualData = (managingOfficer: ManagingOfficerIndividual, moPrivateData) => {
>   for (const managingOfficerData of moPrivateData) {
>     if (managingOfficerData.hashedId === managingOfficer.ch_reference) {
>       managingOfficer.usual_residential_address = mapBOMOAddress(managingOfficerData.residentialAddress);
>       managingOfficer.date_of_birth = mapInputDate(managingOfficerData.dateOfBirth);
>     }
>   }
> };
> ```

Yeah I had considered this, I think I'd brought it up in one of our stand-ups, but I wasn't sure if there was more changes in the pipeline which might mean it changing again, so left it as is.

Reply:
> Just one observation which may or may not help. We need to rely on having a day of birth so that the backend displays data correctly (2 open bugs). So may end up failing if no day found - which would make the day kind of special.

@AndreKramer7654 Can you elaborate on that a bit, sorry? Not sure where that leaves us regarding the above. :-)

Reply:
> We could combine both the `mapMoPrivateAddress` and `mapMoPrivateDOB` functions into one to avoid duplication, something like:
> 
> ```
> const mapManagingOfficerIndividualData = (managingOfficer: ManagingOfficerIndividual, moPrivateData) => {
>   for (const managingOfficerData of moPrivateData) {
>     if (managingOfficerData.hashedId === managingOfficer.ch_reference) {
>       managingOfficer.usual_residential_address = mapBOMOAddress(managingOfficerData.residentialAddress);
>       managingOfficer.date_of_birth = mapInputDate(managingOfficerData.dateOfBirth);
>     }
>   }
> };
> ```

I did talk about this in one of the stand-ups then never did it. I think because the Corporate MO would need its own method then so it was give and take. However I'm going to make the change as it keeps code relative to MO type.

Reply:
Can discuss outside of PR but it may mean we should handle the DOB more separately. But can still do it in a combined method as suggested above.

Reply:
> We could combine both the `mapMoPrivateAddress` and `mapMoPrivateDOB` functions into one to avoid duplication, something like:
> 
> ```
> const mapManagingOfficerIndividualData = (managingOfficer: ManagingOfficerIndividual, moPrivateData) => {
>   for (const managingOfficerData of moPrivateData) {
>     if (managingOfficerData.hashedId === managingOfficer.ch_reference) {
>       managingOfficer.usual_residential_address = mapBOMOAddress(managingOfficerData.residentialAddress);
>       managingOfficer.date_of_birth = mapInputDate(managingOfficerData.dateOfBirth);
>     }
>   }
> };
> ```

I mentioned doing this in Stand-up once but then didn't do it as there would still be a map address method for corporate. I coded it to keep the MO logic organised depending on MO type... bit of work reorganising test files and changing tests :-(

Reply:
This part all looks good to me now, thanks.